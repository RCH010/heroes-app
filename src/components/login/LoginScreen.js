import React, { useContext } from 'react'
import { AuthContext } from '../../auth/AuthContext'
import { types } from '../../types/types'

export const LoginScreen = ({history}) => {

    const {dispatch} = useContext(AuthContext)

    const handleLogin = () => {
        // push, cambia a la pagina, replace sustituye (historia de navegación)
        // history.push('/');
        // history.replace('/');

        const lastPath = localStorage.getItem('lastPath') || '/';


        const actualUser = {
            name: 'Raúl',
            email: 'ra@gm.com'
        };

        const action = {
            type: types.login,
            payload: actualUser
        };
        dispatch(action);
        history.replace(lastPath);
    }

    return (
        <div className='container mt-5'>
            <h1>Login Screen</h1>
            <br/>

            <button
                className= 'btn btn-primary'
                onClick={handleLogin}
            >
                Login
            </button>
        </div>
    )
}
