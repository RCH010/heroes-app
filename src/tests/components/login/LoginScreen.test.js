import { mount } from "enzyme"
import { AuthContext } from "../../../auth/AuthContext"
import { LoginScreen } from "../../../components/login/LoginScreen";
import { types } from "../../../types/types";

describe('Tests in LoginScreen', () => {

    const dispatch = jest.fn();
    const history = {replace: jest.fn()}
    const wrapper = mount(
    <AuthContext.Provider value={{dispatch}}>
        <LoginScreen history={history} />
    </AuthContext.Provider>)

    test('should display correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should call dispatch when loggedin', () => {
        wrapper.find('button').prop('onClick')();
        const actualUser = {
            name: 'Ra√∫l',
            email: 'ra@gm.com'
        };
        const action = {
            type: types.login,
            payload: actualUser
        };
        expect(dispatch).toHaveBeenCalledWith(action);
        expect(history.replace).toHaveBeenCalled();
    })
    
})
