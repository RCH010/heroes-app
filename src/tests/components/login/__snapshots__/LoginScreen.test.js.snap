// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in LoginScreen should display correctly 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <br />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
